app/controllers/admin/tutorials_controller.rb

```diff
@@ -4,15 +4,17 @@ def edit
   end
 
   def create
-    if results.snippet.nil?
-      flash[:error] = 'Sorry, that ID is not valid. Try again?'
-      redirect_to new_admin_youtube_playlist_path
-    else
-      tutorial_from_playlist = Tutorial.create(new_tutorial_params)
-      tutorial_from_playlist.create_playlist_videos
-      context = view_context.link_to('View it here', tutorial_path(tutorial_from_playlist.id))
-      flash[:success] = "Successfully created tutorial. #{context}."
-      redirect_to admin_dashboard_path
+
+    if !missing_params.empty?
+      flash[:error] = "Please enter a #{missing_params}."
+      redirect_to new_admin_tutorial_path
+    elsif valid_thumbnail?(params[:tutorial][:thumbnail])
+      tutorial = Tutorial.create(new_tutorial_params)
+      flash[:success] = "Successfully created tutorial."
+      redirect_to "/tutorials/#{tutorial.id}"
+    else 
+      flash[:error] = "Please enter a valid thumbnail."
+      redirect_to new_admin_tutorial_path
```

Comment:
3 scenarios: if fields are left empty, if the thumbnail doesn't pass the validity test, and if it was successful